ui/apps/platform/src/Containers/Clusters/DelegateScanning/DelegateScanningPage.tsx

```diff
@@ -32,6 +33,8 @@ function DelegateScanningPage() {
         useState<DelegatedRegistryConfig>(initialDelegatedState);
     const [errMessage, setErrMessage] = useState<string>('');
 
+    const { clusters } = useFetchClustersForPermissions(['Deployment']);
```

Comment:
This hook seems like copy-paste from Network Graph.

We assume /main/clusters/delegate-scanning requires Cluster resource with READ_ACCESS, correct?

Therefore, it seems semantically incorrect to the the same hook as dashboard or Network Graph.

Can you write a `useFetchClusters` hook for `fetchClustersAsArray` service function, which will be re-useable for clusters in System Health?

By the way:
1. Clusters list itself needs additional retention info, therefore cannot re-use this hook.
2. Policies fetches a cluster list in 2 places. It deserves follow up with merlin and core-workflows
    * Is Cluster resource with READ_ACCESS required for /main/policy-management/policies because they have global effect?
    * Or does it need to call `useFetchClustersForPermissions` hook?

Reply:
Remind me which resource and access level for /v1/delegatedregistryconfig endpoint?

Reply:
`useFetchClustersForPermissions` was written by backend team specifically to deal with case where limited user could have permission to see the Network Graph, but not any permission on Clusters _per se_.  Therefore, the need to  expose cluster names and IDs for scoping down the network graph without exposing **any** other info about clusters. Principle of least privilege, in the wild.

At this point, I'll confirm the permissions needed for this feature, and adjust getting the clusters as needed. I don't want to solve for the general case with another new hook, especially if we only need the names/IDs for the functionality on this page, and we already have the `useFetchClustersForPermissions` hook to give us that.

Reply:
@rhybrillou can you confirm whether empty array is correct instead of `['Deployment']` arg in this situation?

https://github.com/stackrox/stackrox/blob/master/proto/api/v1/role_service.proto#L274-L275

```
    // If no permission is given in input, all clusters allowed by the requester scope for
    // any permission with cluster scope or narrower will be part of the response.
```

Reply:
Just heard from David Caravello that this will require the "Administration" permission.

However,
```
const { clusters } = useFetchClustersForPermissions(['Administration']);
```
returns an empty cluster list, and no error or other information about why it's empty.

I think the endpoint `/v1/sac/clusters?permissions=` is too bespoke for its one use case so far, to be useful as a public API. We need to come to a better understanding of what it's available for and what its "permissions" query param accepts. (The API doc for this input just list the vague "permissions | Array of strings".)

In this page, I only need what it returns, cluster names and IDs, but I can fall back to the full cluster endpoint since "Administration" permission implies that we can also require "Cluster" permission.  (Does Administration permission automatically give Cluster permission, or could a user configure a permission set without it?)

cc: @rukletsov

Reply:
https://github.com/stackrox/stackrox/blob/master/proto/api/v1/role_service.proto#L272

> // If the input only contains permissions at global level, the output will be an empty list.